### DISCUSSION TITLE
How do I manage state with getServerSideProps()?

### DISCUSSION THREAD

**Discussion by tar-aldev** [vercel/next.js#22103] [Help]
Let's say I have `getServerSideProps` function. There I am fetching a list of blog posts and they are shown on the screen.
A user can delete the post or mark it as favorite.
In react I would have initial state with`posts = []` and then just update it.
In next, looks like I cannot do it this way, because server side props cannot be mutated.
So when I load the posts and then delete one of them, how do I update my page, to not show this post?
Or another example is pagination:

Let's say I pre rendered my posts first page. Then I want to load posts from the second page, but again cannot change what was passed as serverSideProps.

So how do I deal with it?
Should I put these props into component's state and then update it? As far as I know it is an anipattern in React world.
Any help would be appreciated.

**Comment by jamesmosier** [comment]
> Should I put these props into component's state and then update it?

Exactly! You can use `getServerSideProps` as an "initial state" and then manage the state locally in the component. 

```js
function MyComponent({ initialPosts }) {
  const [posts, setPosts] = useState(initialPosts) // set the initial state in the component 
}

export async function getServerSideProps() {
  return { props: { initialPosts: [/* your posts... */] } } 
}
```

You could also use something like `swr` in your component to hydrate the initial state and then update it as needed on the client side: https://swr.vercel.app/docs/with-nextjs#pre-rendering

**Reply by tar-aldev** [reply]
Thank you so much! I really appreciate that next.js team is always willing to help )

**Reply by rohilpinto** [reply]
Thank you

**Reply by alexalannunes** [reply]
thanks
